hyperspy/drawing/_markers/rectangle.py

```diff
@@ -0,0 +1,94 @@
+# -*- coding: utf-8 -*-
+# Copyright 2007-2011 The Hyperspy developers
+#
+# This file is part of  Hyperspy.
+#
+#  Hyperspy is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+#  Hyperspy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with  Hyperspy.  If not, see <http://www.gnu.org/licenses/>.
+
+import matplotlib.pyplot as plt
+
+from hyperspy.drawing.marker import MarkerBase
+
+
+class Rectangle(MarkerBase):
+    """Rectangle marker that can be added to the signal figure
+
+    Parameters
+    ---------
+    x1: array or float
+        The position of the up left corner of the rectangle in x.
+        If float, the marker is fixed.
+        If array, the marker will be updated when navigating. The array should
+        have the same dimensions than the nagivation axes.
+    y1: array or float
+        The position of the up left corner of the rectangle in y.
+        see x1 arguments
+    x2: array or float
+        The position of the down right corner of the rectangle in x.
+        see x1 arguments
+    y2: array or float
+        The position of the up down right of the rectangle in y.
```

Comment:
..of the __~~up~~__ down right..

I would also probably use "top" and "bottom" instead of "up" and "down"